### PR TITLE
Add getRect() and isVisible()

### PR DESCRIPTION
**Pull request by mrmacete**
- the rect is in a CGRect friendly format

- the visibility query is based on heuristics
from http://stackoverflow.com/a/15203639/5418401

### CODE REVIEW

**Comment by oleavr** [comment]
w00t!

File: index.js [hunk 1]
```diff
@@ -242,6 +252,49 @@ TAP_ELEMENT_SCRIPT = `function tapElement(params) {
...
+  // Return true if any of its four corners are visible
+  return {
+    visible: (
+      element.contains(efp(rect.left, rect.top)) ||
+      element.contains(efp(rect.right, rect.top)) ||
+      element.contains(efp(rect.right, rect.bottom)) ||
+      element.contains(efp(rect.left, rect.bottom))
+    )
+  };
+}`;
```
Comment (oleavr) [inline index.js]: Can't we return a boolean here instead? (Or do the unboxing on the other side if not.)

File: index.js [hunk 2]
```diff
@@ -242,6 +252,49 @@ TAP_ELEMENT_SCRIPT = `function tapElement(params) {
...
+
+/*
+ * Adapted from: http://stackoverflow.com/a/15203639/5418401
+ */
+IS_ELEMENT_VISIBLE_SCRIPT = `function isElementVisible(params) {
+  var element = window._fridaElementByRef[params.ref];
+  var rect = element.getBoundingClientRect();
+  if (rect.width == 0 || rect.height == 0) {
```
Comment (oleavr) [inline index.js]: Would be good to use strict equality checks here.

File: index.js (hunk 1 above)
Comment (mrmacete) [inline index.js]: yes, i'll go with unboxing because `perform()` treats the return value as an object

Comment (oleavr) [inline index.js]: Sweet.

**Comment by oleavr** [comment]
+ frida-uiwebview@1.1.0

**Pull request merged**